spec/formatting.md

```diff
@@ -18,3 +18,236 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is not well-formed.
+
+  Example invalid messages resulting in a Syntax error:
+
+  ```
+  {Missing end brace
+  ```
+
+  ```
+  {Unknown {?placeholder}}
+  ```
+
+  ```
+  let $var = {(no message body)}
+  ```
+
+- **Data Model errors** occur when a message is invalid due to
+  violating one of the following semantic requirements on its structure:
```

Comment:
Proposal:
> violating one of the following semantic requirements
to:
> violating one of the semantic requirements

Reason: we didn't yet carve in stone the exact semantic requirements
If we say "the following" then the expectation is that this section will list all of them.
So any future update in semantic requirements would require updating this section.

By taking out "following" we transform the section below in a non-exhaustive list of examples.
The same way we do for other error types.

Reply:
```suggestion
  violating one of the semantic requirements on its structure:
```